span_sinks.go

```diff
@@ -0,0 +1,350 @@
+package veneur
+
+import (
+	"container/ring"
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/Sirupsen/logrus"
+	lightstep "github.com/lightstep/lightstep-tracer-go"
+	opentracing "github.com/opentracing/opentracing-go"
+	"github.com/opentracing/opentracing-go/ext"
+	"github.com/stripe/veneur/ssf"
+	"github.com/stripe/veneur/trace"
+)
+
+const datadogResourceKey = "resource"
+const datadogNameKey = "name"
+
+const lightStepOperationKey = "name"
+
+const totalSpansFlushedMetricKey = "worker.spans_flushed_total"
+
+// SpanSink is a receiver of spans that handles sending those spans to some
+// downstream sink. Calls to `Ingest(span)` are meant to give the sink control
+// of the span, with periodic calls to flush as a signal for sinks that don't
+// handle their own flushing in a separate goroutine, etc.
+type SpanSink interface {
```

Comment:
Does this need to be exported? Advantage of keeping it internal (for now): it's easy to export it later and doesn't require a major version bump.